tools/nimgrep.nim

```diff
@@ -851,6 +897,8 @@ iterator processFile(searchOptC: SearchOptComp[Pattern], filename: string,
   var error = false
   if optFilenames in options:
     buffer = filename
+  elif optPipe in options:
+    buffer = stdin.readAll()
```

Comment:
that's the thing I was complaining about above; one of the main points of `stdin` is to allow streaming, eg so that
* you don't need to hold in memory the whole input
* you can start processing right away

feel free to leave this comment as unaddressed to be left for future work

Reply:
It's incompatible with multi-line patterns.
So we need to add option for switching to one-line patterns to go to streaming mode.
And it's incompatible with options like context.
BTW when GNU grep switches to PCRE (for multi-line patterns), it also starts to read buffer by whole.

What's so great about the streaming mode? Memory consumption, more parallel execution, ...?

Reply:
added to limitations.